app/api/v2/managers/operation_api_manager.py

```diff
@@ -19,4 +91,52 @@ async def get_operation_object(self, operation_id: str, access: dict):
             raise JsonHttpNotFound(f'Operation not found: {operation_id}')
         if operation.match(access):
             return operation
-        raise JsonHttpForbidden(f'Cannot view operation due to insufficient permissions: {operation_id}')
+        raise JsonHttpForbidden(f'Insufficient permissions to view operation {operation_id}')
+
+    def search_operation_for_link(self, operation: Operation, link_id: str):
+        for link in operation.chain:
+            if link.id == link_id:
+                return link
+        raise JsonHttpNotFound(f'Link {link_id} was not found in Operation {operation.id}')
+
+    def validate_link_data(self, link_data: dict):
+        if not link_data.get('executor'):
+            raise JsonHttpBadRequest('\'executor\' is a required field for link creation.')
+        if not link_data['executor'].get('name'):
+            raise JsonHttpBadRequest('\'name\' is a required field for link executor.')
+        if not link_data['executor'].get('command'):
+            raise JsonHttpBadRequest('\'command\' is a required field for link executor.')
+        if not link_data.get('paw'):
+            raise JsonHttpBadRequest('\'paw\' is a required field for link creation.')
+
+    async def get_agent(self, access: dict, data: dict):
```

Comment:
We should only be able to assign potential links to agents that are part of the operation, so we should also check to make sure that the agent is part of the operation. Rather than doing a data svc lookup, we could check operation.agents and return the agent that has the matching paw